src/engraving/rendering/score/tdraw.cpp

```diff
@@ -2176,12 +2176,25 @@ void TDraw::draw(const MeasureRepeat* item, Painter* painter)
             pen.setWidthF(hBarThickness);
             painter->setPen(pen);
 
-            double twoMeasuresWidth = 2 * item->measure()->width();
+            double twoMeasuresLeftWidth = item->measure()->width() + item->measure()->prevMeasure()->width();
+            if (item->measure()->prevMeasure()->isFirstInSystem()) {
+                //twoMeasuresLeftWidth -= yyy; // account for clef, time- and key sig
+            }
+            if (item->measure()->prevMeasure()->repeatStart()) {
+                //twoMeasuresLeftWidth -= xxx; // account for repeat barline
+            }
+            double twoMeasuresRightWidth = item->measure()->nextMeasure()->width() + item->measure()->nextMeasure()->nextMeasure()->width();
+            if (item->measure()->nextMeasure()->nextMeasure()->repeatEnd()) {
+                //twoMeasuresRightWidth -= xxx; // account fot repeat barline
+            }
+            if (item->measure()->nextMeasure()->nextMeasure()->isLastInSystem()) {
+                //twoMeasuresRightWidth -= yyy; // account for courtesy clef, time and key signature
+            }
```

Comment:
These types of calculations should never _ever_ be done in the drawing code, because it's code that gets executed at every frame refresh of the screen. Ideally, the drawing code should not do a single calculation, but literally just take the info that's already there and draw it. These calculations belong in the layout code. Just mentioning this because I've seen a lot of these errors.